src/WpBookingsFrontUi.php

```diff
@@ -1,144 +1,256 @@
-     */
-    protected function _getRedirectUrl()
-    {
-        return get_permalink($this->cartPageId);
-    }
-
-    /**
-     * Get API base url.
-     * 
-     * @since [*next-version*]
-     * 
-     * @return string
-     */
-    protected function _getApiBaseUrl()
-    {
-        return rest_url($this->apiBaseUrl);
-    }
-
-    /**
-     * Add WP styles and scripts enqueuing.
-     *
-     * @since [*next-version*]
-     */
-    public function enqueueAssets($c)
-    {
-        wp_enqueue_script(RC_WP_BOOKINGS_FRONT_UI_MODULE_KEY . '-main', $c->get('bookings_front_ui/main_script'), [], false, true);
-        wp_enqueue_script(RC_WP_BOOKINGS_FRONT_UI_MODULE_KEY, $c->get('bookings_front_ui/script'), [], false, true);
-        wp_enqueue_style(RC_WP_BOOKINGS_FRONT_UI_MODULE_KEY, $c->get('bookings_front_ui/style'));
-    }
-}
\ No newline at end of file
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Data\Container\ContainerFactoryInterface;
+use Dhii\Event\EventFactoryInterface;
+use Psr\Container\ContainerInterface;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\Bookings\WordPress\Module\Handlers\AssetsEnqueueHandler;
+use RebelCode\Bookings\WordPress\Module\Handlers\MainComponentHandler;
+use RebelCode\Bookings\WordPress\Module\Handlers\OutputTemplateHandler;
+use RebelCode\Bookings\WordPress\Module\Handlers\StateEnqueueHandler;
+use RebelCode\Modular\Module\AbstractBaseModule;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Dhii\Output\PlaceholderTemplateFactory;
+use Dhii\Output\TemplateFactoryInterface;
+
+class WpBookingsFrontUi extends AbstractBaseModule
+{
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable         $key                  The module key.
+     * @param string[]|Stringable[]     $dependencies         The module  dependencies.
+     * @param ContainerFactoryInterface $configFactory        The config factory.
+     * @param ContainerFactoryInterface $containerFactory     The container factory.
+     * @param ContainerFactoryInterface $compContainerFactory The composite container factory.
+     * @param EventManagerInterface     $eventManager         The event manager.
+     * @param EventFactoryInterface     $eventFactory         The event factory.
+     */
+    public function __construct(
+        $key,
+        $dependencies,
+        $configFactory,
+        $containerFactory,
+        $compContainerFactory,
+        $eventManager,
+        $eventFactory
+    ) {
+        $this->_initModule($key, $dependencies, $configFactory, $containerFactory, $compContainerFactory);
+        $this->_initModuleEvents($eventManager, $eventFactory);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function setup()
+    {
+        return $this->_setupContainer(
+            $this->_loadPhpConfigFile(WP_BOOKINGS_FRONT_UI_MODULE_CONFIG), [
```

Comment:
The square brace on this line is confusing: it looks as if the two statements are related.